packages/blue-sdk-ethers-liquidation/src/tokens/usual.ts

```diff
@@ -0,0 +1,181 @@
+import { AbstractSigner, MaxUint256, Provider } from "ethers";
+import { LiquidationEncoder } from "../LiquidationEncoder";
+import { curvePools } from "../addresses";
+import { mainnetAddresses } from "../addresses";
+import {
+  encodeCurveSwap,
+  encodeRemoveLiquidityFromCurvePool,
+  getCurveSwapInputAmountFromOutput,
+  getCurveSwapOutputAmountFromInput,
+  getCurveWithdrawalAmount,
+} from "../swap/curve";
+
+export const USD0_USDC_USD0_INDEX = 0;
+export const USD0_USDC_USDC_INDEX = 1;
+
+export const USD0_USD0PP_USD0_INDEX = 0;
+export const USD0_USD0PP_USDPP_INDEX = 1;
+
+/**
+ *  Swaps USD0USD0++ for USDC through the USD0/USD0++ && USD0/USDC pools
+ *  Route is USD0USD0++ -> USD0 -> USDC
+ * @returns the total swapped USDC amount
+ */
+export async function swapUsd0Usd0PPToUSDC(
+  provider: AbstractSigner<Provider>,
```

Comment:
The encoder should expose the underlying `signer` & `provider`, which removes the possibility of making this function fail because of incompatible provider

Also, you could move this function inside the extended encoder and name it more generally `curveSwapUsd0Usd0PPForUsdc` (or to, idrc)